tools/nblIntrinicGenerator/example input/spirv.type_mappings.json

```diff
@@ -0,0 +1,59 @@
+{
+    "default": "uint32_t",
+    "classMappings":[
+        {
+            "class": "Atomic",
+            "types": [
+                {
+                    "kind": "IdResult",
+                    "type": "T"
+                },
+                {
+                    "kind": "IdRef",
+                    "name": "Pointer",
+                    "type": "P",
+                    "attributes": [
+                        "vk::ext_reference"
+                    ]
```

Comment:
vk::ext_reference is actually the only attribute we can apply.

Btw we need to modify how we codegen, the reason is that `[vk::ext_reference] P` doesnt work in all cases, so we need BOTH
```cpp
template<typename T>
[[vk::ext_instruction(spv::OpAtomicLoad)]]
T AtomicLoad([[vk::ext_reference]] T pointer, uint32_t memory, uint32_t semantics);

template<typename T, typename Ptr_T>
[[vk::ext_instruction(spv::OpAtomicLoad)]]
T AtomicLoad(Ptr_T pointer, uint32_t memory, uint32_t semantics);
```

Reply:
obviously if you have a lot of `&` this turns into combinatorial explosion of overloads `2^parameters_with_overloads` but thats the fault of https://github.com/microsoft/DirectXShaderCompiler/issues/6578